src/EnergyPlus/FileSystem.cc

```diff
@@ -223,7 +223,7 @@ namespace FileSystem {
 #endif
 
         extension.remove_prefix(extension.find_last_of('.') + 1);
-        return static_cast<FileTypes>(getEnumerationValue(FileTypesExt, extension));
+        return static_cast<FileTypes>(getEnumerationValueUpper(FileTypesExt, extension));
```

Comment:
That's not the way to do this. The way to do it is to make a `FileTypeExtNamesUC` array where the extensions are upper-case and then to call `getEnumerationValue` with that and `MakeUPPERCase(extension)`. This way, you call `MakeUPPERCase` exactly once. The way you've implemented things here `MakeUPPERCase gets called *twice on every comparison*.

Reply:
One more thing is that you should move `FileTypesExt` and `FileTypesExtUC` to the .cc file since they are only used in this module.